### PR TITLE
Refactoring Symmetry Segmentation of An Le's project in gsoc2017

### PR DESCRIPTION
**Pull request by anindex**
###Summary
Refactoring these annotators:
- OverSegmentationAnnotator
- RotationalSymmetryAnnotator
- RotationalSymmetrySegmentation
- BilateralSymmetryAnnotator
- BilateralSymmetrySegmentation

Other modifications: 
- utils related to robosherlock Graph libraries
- add option to use non-NAN cloud or not in PlaneAnnotator

### CODE REVIEW

**Comment by bbferka** [comment]
@anindex the fail on travis was reproducible on my machine, by simply running catkin run_tests; Even though I don't see what you could have touched here that makes the utest fail, on master they run through fine.

File: descriptors/annotators/detection/PlaneAnnotator.xml [hunk 1]
```diff
@@ -72,6 +80,13 @@
         </value>
       </nameValuePair>
 
+      <nameValuePair>
+        <name>use_non_nan_cloud</name>
+        <value>
+          <string>false</string>
```
Comment (bbferka) [inline descriptors/annotators/detection/PlaneAnnotator.xml]: found the probelm; this should be <boolean> tag not <string>

Comment (anindex) [reply descriptors/annotators/detection/PlaneAnnotator.xml]: yeah this is a hilarious mistake :). Should be tag boolean

Comment (bbferka) [reply descriptors/annotators/detection/PlaneAnnotator.xml]: I fixed

**Comment by bbferka** [comment]
@anindex ok, this looks better now; Maybe we should put your unit tests back? Can your alogrithms run on the data in the sample folder? I would like to keep the amount of raw data in the repo at a minimum

**Pull request merged**